tests/cupyx_tests/scipy_tests/sparse_tests/test_index.py

```diff
@@ -0,0 +1,110 @@
+import unittest
+
+import cupy
+from cupy import sparse
+from cupy import testing
+
+import numpy
+
+import pytest
+
+
+@testing.parameterize(*testing.product({
+    'format': ['csr', 'csc'],
+    'density': [0.1, 0.4, 0.9],
+    'dtype': ['float32', 'float64', 'complex64', 'complex128'],
+    'n_rows': [25, 150],
+    'n_cols': [25, 150]
+}))
+@testing.with_requires('scipy')
+@testing.gpu
+class TestIndexing(unittest.TestCase):
+
+    def _run(self, maj, min=None):
+        a = sparse.random(self.n_rows, self.n_cols,
+                          format=self.format,
+                          density=self.density)
+
+        # sparse.random doesn't support complex types
+        # so we need to cast
+        a = a.astype(self.dtype)
+
+        maj_h = maj.get() if isinstance(maj, cupy.ndarray) else maj
+        min_h = min.get() if isinstance(min, cupy.ndarray) else min
+
+        expected = a.get()
+
+        if min is not None:
+            expected = expected[maj_h, min_h]
+            actual = a[maj, min]
+        else:
+            expected = expected[maj_h]
+            actual = a[maj]
+
+        if sparse.isspmatrix(actual):
+            actual.sort_indices()
+            expected.sort_indices()
+
+            cupy.testing.assert_array_equal(
+                actual.indptr, expected.indptr)
+            cupy.testing.assert_array_equal(
+                actual.indices, expected.indices)
+            cupy.testing.assert_array_equal(
+                actual.data, expected.data)
+        else:
+            cupy.testing.assert_array_equal(
+                actual.ravel(), numpy.array(expected).ravel())
```

Comment:
This should work. CuPy internally does the copy before the comparison.
```suggestion
            testing.assert_array_equal(actual, expected)
```